From 617fc3881c5e66a8c1763359e8a11f07f1332fd5 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:55:24 -0400
Subject: [fs] ext4: avoid trying to kfree an ERR_PTR pointer

Message-id: <1414058125-4183-171-git-send-email-lczerner@redhat.com>
Patchwork-id: 98247
O-Subject: [RHEL 7.1 PATCH v2 170/171] ext4: avoid trying to kfree an ERR_PTR pointer
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Theodore Ts'o <tytso@mit.edu>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit a9cfcd63e8d206ce4235c355d857c4fbdf0f4587

 Thanks to Dan Carpenter for extending smatch to find bugs like this.
 (This was found using a development version of smatch.)

 Fixes: 36de928641ee48b2078d3fe9514242aaa2f92013
 Reported-by: Dan Carpenter <dan.carpenter@oracle.com
 Signed-off-by: Theodore Ts'o <tytso@mit.edu>
 Cc: stable@vger.kernel.org
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/namei.c b/fs/ext4/namei.c
index 62237d9..0d93bc5 100644
--- a/fs/ext4/namei.c
+++ b/fs/ext4/namei.c
@@ -3163,6 +3163,7 @@ static int ext4_rename(struct inode *old_dir, struct dentry *old_dentry,
      &new.de, &new.inlined);
  if (IS_ERR(new.bh)) {
   retval = PTR_ERR(new.bh);
+  new.bh = NULL;
   goto end_rename;
  }
  if (new.bh) {
diff --git a/fs/ext4/resize.c b/fs/ext4/resize.c
index f3b84cd..3702801 100644
--- a/fs/ext4/resize.c
+++ b/fs/ext4/resize.c
@@ -572,6 +572,7 @@ handle_bb:
   bh = bclean(handle, sb, block);
   if (IS_ERR(bh)) {
    err = PTR_ERR(bh);
+   bh = NULL;
    goto out;
   }
   overhead = ext4_group_overhead_blocks(sb, group);
@@ -600,6 +601,7 @@ handle_ib:
   bh = bclean(handle, sb, block);
   if (IS_ERR(bh)) {
    err = PTR_ERR(bh);
+   bh = NULL;
    goto out;
   }
 
-- 
1.7.1